otherlibs/unix/unixsupport.h

```diff
@@ -58,10 +58,11 @@ struct filedescr {
 #define FLAGS_FD_IS_BLOCKING (1<<0)
 
 
-#define Handle_val(v) (((struct filedescr *) Data_custom_val(v))->fd.handle)
-#define Socket_val(v) (((struct filedescr *) Data_custom_val(v))->fd.socket)
-#define Descr_kind_val(v) (((struct filedescr *) Data_custom_val(v))->kind)
-#define Flags_fd_val(v) (((struct filedescr *) Data_custom_val(v))->flags_fd)
+#define Descr_val(v) ((struct filedescr *) Data_custom_val(v))
```

Comment:
Since the result of this points inside the GC object `v`, the resulting value is only valid until the next GC point and is entirely invalid inside a blocking section.